File: SoftLogic.Core/Win/Reporting/SPReportManager.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;
using System.ComponentModel;
using SoftLogik.Win.UI;


namespace SoftLogik.Win
{
	namespace Reporting
	{
		public partial class SPReportManager
		{
			
			
			private static SPReportViewer _reportViewer = null;
			private static SPReportSettings _reportSettings = null;
			private static Form _MdiParent = null;
			
			private static SPReportViewer Viewer
			{
				get
				{
					if (_reportViewer == null)
					{
						_reportViewer = new SPReportViewer();
					}
					if (_reportViewer.IsDisposed)
					{
						_reportViewer = new SPReportViewer();
					}
					return _reportViewer;
				}
			}
			public static SPReportSettings Settings
			{
				get
				{
					if (_reportSettings == null)
					{
						_reportSettings = new SPReportSettings();
					}
					if (_reportSettings.IsDisposed)
					{
						_reportSettings = new SPReportSettings();
					}
					return _reportSettings;
				}
			}
			
			
			public static Form MdiParent
			{
				set
				{
					_MdiParent = value;
				}
			}
			
			[Description("Load and Show a Report a specified Report File in the Customized Report Viewer.")]public static void ShowReport(string ReportFile)
			{
				Settings.MdiParent = _MdiParent;
				Settings.ViewerForm = Viewer;
				Settings.Show();
			}
			
			[Description("Load and Print a Report a specified Report File.")]public static void PrintReport(string ReportFile)
			{
				Viewer.MdiParent = _MdiParent;
				Settings.MdiParent = _MdiParent;
				Settings.ViewerForm = Viewer;
			}
			
		}
	}
	
	
}

File: SPCode/Reporting/SPReportManager.vb
Imports System.ComponentModel
Imports SoftLogik.Win.UI

Namespace Reporting
    Public Class SPReportManager

        Private Shared WithEvents _reportViewer As SPReportViewer = Nothing
        Private Shared WithEvents _reportSettings As SPReportSettings = Nothing
        Private Shared _MdiParent As Form = Nothing

        Private Shared ReadOnly Property Viewer() As SPReportViewer
            Get
                If _reportViewer Is Nothing Then _reportViewer = New SPReportViewer
                If _reportViewer.IsDisposed Then _reportViewer = New SPReportViewer
                Return _reportViewer
            End Get
        End Property
        Public Shared ReadOnly Property Settings() As SPReportSettings
            Get
                If _reportSettings Is Nothing Then _reportSettings = New SPReportSettings
                If _reportSettings.IsDisposed Then _reportSettings = New SPReportSettings
                Return _reportSettings
            End Get
        End Property


        Public Shared WriteOnly Property MdiParent() As Form
            Set(ByVal value As Form)
                _MdiParent = value
            End Set
        End Property

        <Description("Load and Show a Report a specified Report File in the Customized Report Viewer.")> _
        Public Shared Sub ShowReport(ByVal ReportFile As String)
            With Settings
                .MdiParent = _MdiParent
                .ViewerForm = Viewer
                .Show()
            End With
        End Sub

        <Description("Load and Print a Report a specified Report File.")> _
        Public Shared Sub PrintReport(ByVal ReportFile As String)
            Viewer.MdiParent = _MdiParent
            With Settings
                .MdiParent = _MdiParent
                .ViewerForm = Viewer
            End With
        End Sub

    End Class
End Namespace

